app/src/main/java/com/twilio/video/app/ui/room/RoomActivity.java

```diff
@@ -281,32 +180,48 @@ protected void onCreate(Bundle savedInstanceState) {
         // Setup toolbar
         setSupportActionBar(toolbar);
 
-        // Setup Audio
-        audioManager = (AudioManager) getSystemService(Context.AUDIO_SERVICE);
-        audioManager.setSpeakerphoneOn(true);
-        savedVolumeControlStream = getVolumeControlStream();
-
         // setup participant controller
         participantController = new ParticipantController(thumbnailLinearLayout, primaryVideoView);
         participantController.setListener(participantClickListener());
 
+        displayName = sharedPreferences.getString(Preferences.DISPLAY_NAME, "");
+
         // Setup Activity
         statsScheduler = new StatsScheduler();
-        obtainVideoConstraints();
         requestPermissions();
     }
 
     @Override
     protected void onResume() {
         super.onResume();
-        boolean isAppLinkProvided = checkIntentURI();
-        displayName = sharedPreferences.getString(Preferences.DISPLAY_NAME, null);
-        updateUi(room, isAppLinkProvided);
+        //        boolean isAppLinkProvided = checkIntentURI();
+        //        updateUi(isAppLinkProvided);
         restoreCameraTrack();
-        initializeRoom();
         updateStats();
     }
 
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        roomManager.processViewEvent(TearDownLocalMedia.INSTANCE);
+    }
+
+    private void bindViewState(@Nullable RoomViewState viewState) {
+        updateUi(viewState);
+    }
+
+    private void bindViewEffects(@Nullable RoomViewEffect viewEffect) {
+        if (viewEffect != null) {
+            if (viewEffect instanceof ScreenShareError) {
+                Snackbar.make(primaryVideoView, R.string.screen_capture_error, Snackbar.LENGTH_LONG)
+                        .show();
+            }
+            if (viewEffect instanceof RequestScreenSharePermission) {
+                //                requestScreenCapturePermission();
```

Comment:
Yeah same thing here. We should either cut tickets or track these in a PR description checklist so what needs to be completed for this work to be merged.